hack/make-rules/update.sh

```diff
@@ -78,3 +78,12 @@ done
 ${KUBE_ROOT}/hack/arktos_copyright.sh ${KUBE_ROOT} ${KUBE_ROOT}/_output
 
 echo -e "${color_green:?}Update scripts completed successfully${color_norm}"
+
+echo "Now let's check whether there is file updated."
+echo "We define return value as 0 if there is no file updated, and 1 if there is file updated."
+MODIFIED_FILES=$(git status | grep modified:) || true
```

Comment:
we should only do this in ci builds. normal dev environment builds should not because users can have filed checked out or staged.

Reply:
Updated. Only check in ci environment.

Reply:
This should work in CI the same way as work in dev env. Otherwise, dev will be confused if CI fails but local dev make update pass

Reply:
Now changed back to check in both cases.